### ISSUE TITLE
AttributeError: 'list' object has no attribute 'startswith'

### ISSUE THREAD

**Issue by WorstCodeWay** [ultralytics/yolov5#4876]
Before submitting a bug report, please be aware that your issue **must be reproducible** with all of the following,
otherwise it is non-actionable, and we can not help you:

- **Current repo**: run `git fetch && git status -uno` to check and `git pull` to update repo
- **Common dataset**: coco.yaml or coco128.yaml
- **Common environment**: Colab, Google Cloud, or Docker image. See https://github.com/ultralytics/yolov5#environments

If this is a custom dataset/training question you **must include** your `train*.jpg`, `val*.jpg` and `results.png`
figures, or we can not help you. You can generate these with `utils.plot_results()`.

## 🐛 Bug

```
Traceback (most recent call last):
  File "train.py", line 611, in <module>
    main(opt)
  File "train.py", line 509, in main
    train(opt.hyp, opt, device, callbacks)
  File "train.py", line 85, in train
    loggers = Loggers(save_dir, weights, opt, hyp, LOGGER)  # loggers instance
  File "/home/user/Documents/vist/deeplearning/yolov5/utils/loggers/__init__.py", line 62, in __init__
    self.wandb = WandbLogger(self.opt, run_id)
  File "/home/user/Documents/vist/deeplearning/yolov5/utils/loggers/wandb/wandb_utils.py", line 174, in __init__
    self.data_dict = check_wandb_dataset(opt.data)
  File "/home/user/Documents/vist/deeplearning/yolov5/utils/loggers/wandb/wandb_utils.py", line 46, in check_wandb_dataset
    is_wandb_artifact = (data_dict['train'].startswith(WANDB_ARTIFACT_PREFIX) or
AttributeError: 'list' object has no attribute 'startswith'
```

## To Reproduce (REQUIRED)

Input:
**data.yaml**
```
# YOLOv5 🚀 by Ultralytics, GPL-3.0 license
#  dataset
# Example usage: python train.py --data data.yaml

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: /home/user/Downloads/datasets/data_only # dataset root dir
train:  # train images (relative to 'path')
  - 20210729/images
  - 20210624/images
val:   # train images (relative to 'path') 5000 images
  - 20210624/images_depth
test: images  # 20288 of 40670 images, submit to https://competitions.codalab.org/competitions/20794

# Classes
nc: 1  # number of classes
names: ['someclass']  # class names


```

Output:

```
python train.py --data data.yaml --weights yolov5s.pt --batch-size 32 --epochs 1000
train: weights=yolov5s.pt, cfg=, data=pandents.yaml, hyp=data/hyps/hyp.scratch.yaml, epochs=1000, batch_size=32, imgsz=640, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, adam=False, sync_bn=False, workers=8, project=runs/train, entity=None, name=exp, exist_ok=False, quad=False, linear_lr=False, label_smoothing=0.0, upload_dataset=False, bbox_interval=-1, save_period=-1, artifact_alias=latest, local_rank=-1, freeze=0, patience=100
github: skipping check (offline), for updates see https://github.com/ultralytics/yolov5
YOLOv5 🚀 v5.0-414-g0d8a184 torch 1.8.1+cu102 CUDA:0 (GeForce RTX 2070 with Max-Q Design, 7982.3125MB)

hyperparameters: lr0=0.01, lrf=0.2, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0
TensorBoard: Start with 'tensorboard --logdir runs/train', view at http://localhost:6006/
wandb: Currently logged in as: aisoon (use `wandb login --relogin` to force relogin)
wandb: wandb version 0.12.2 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade
wandb: Tracking run with wandb version 0.12.1
wandb: Syncing run zesty-glitter-27
wandb: ⭐️ View project at https://wandb.ai/aisoon/YOLOv5
wandb: 🚀 View run at https://wandb.ai/aisoon/YOLOv5/runs/2jj5cqo8
wandb: Run data is saved locally in /home/jason/Documents/vist/deeplearning/yolov5/wandb/run-20210921_205444-2jj5cqo8
wandb: Run `wandb offline` to turn off syncing.

Traceback (most recent call last):
  File "train.py", line 611, in <module>
    main(opt)
  File "train.py", line 509, in main
    train(opt.hyp, opt, device, callbacks)
  File "train.py", line 85, in train
    loggers = Loggers(save_dir, weights, opt, hyp, LOGGER)  # loggers instance
  File "/home/jason/Documents/vist/deeplearning/yolov5/utils/loggers/__init__.py", line 62, in __init__
    self.wandb = WandbLogger(self.opt, run_id)
  File "/home/jason/Documents/vist/deeplearning/yolov5/utils/loggers/wandb/wandb_utils.py", line 174, in __init__
    self.data_dict = check_wandb_dataset(opt.data)
  File "/home/jason/Documents/vist/deeplearning/yolov5/utils/loggers/wandb/wandb_utils.py", line 46, in check_wandb_dataset
    is_wandb_artifact = (data_dict['train'].startswith(WANDB_ARTIFACT_PREFIX) or
AttributeError: 'list' object has no attribute 'startswith'

wandb: Waiting for W&B process to finish, PID 7008
wandb: Program failed with code 1.  Press ctrl-c to abort syncing.
wandb:                                                                                
wandb: Find user logs for this run at: /home/jason/Documents/vist/deeplearning/yolov5/wandb/run-20210921_205444-2jj5cqo8/logs/debug.log
wandb: Find internal logs for this run at: /home/jason/Documents/vist/deeplearning/yolov5/wandb/run-20210921_205444-2jj5cqo8/logs/debug-internal.log
wandb: Synced 5 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
wandb: 
wandb: Synced zesty-glitter-27: https://wandb.ai/aisoon/YOLOv5/runs/2jj5cqo8
```

## Environment

- OS: Ubuntu 18.04
- GPU 2070

**Comment by glenn-jocher** [comment]
@WorstCodeWay thanks for the bug report! Maybe our W&B expert @AyushExel can help with this.

We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

* ✅ **Minimal** – Use as little code as possible that still produces the same problem
* ✅ **Complete** – Provide **all** parts someone else needs to reproduce your problem in the question itself
* ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

In addition to the above requirements, for [Ultralytics](https://ultralytics.com/) to provide assistance your code should be:

* ✅ **Current** – Verify that your code is up-to-date with current GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been resolved by previous commits.
* ✅ **Unmodified** – Your problem must be reproducible without any modifications to the codebase in this repository. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all of the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) and providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by AyushExel** [comment]
I'm looking into this. I'll push a fix soon

**Comment by AyushExel** [comment]
@glenn-jocher @WorstCodeWay Fix -> https://github.com/ultralytics/yolov5/pull/4879

**Comment by glenn-jocher** [comment]
@WorstCodeWay good news 😃! Your original issue may now be fixed ✅ in PR #4879. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
